File: javascript/src/integrationSrc.js
function add(a, b) {
    return a + b;
}

function subtract(a, b) {
    return a - b;
}

function multiply(a, b) {
    return a * b;
}

function divide(a, b) {
    if (b === 0) {
        throw new Error("Division by zero is not allowed.");
    }
    return a / b;
}

function complexMathCalculation(a, b, c, d) {
    const sum = add(a, b);
    const difference = subtract(c, d);
    const product = multiply(sum, difference);
    const result = divide(product, 2);
    let finalResult = result;
    
    for (let i = 0; i < 5; i++) {
        if (i % 2 === 0) {
            finalResult = add(finalResult, i);
        } else {
            finalResult = subtract(finalResult, i);
        }
    }
    return finalResult;
}

// mutant ipex
function complexMathCalculation1(d, b, c, a) {
    const sum = add(a, b);
    const difference = subtract(c, d);
    const product = multiply(sum, difference);
    const result = divide(product, 2);
    let finalResult = result;
    
    for (let i = 0; i < 5; i++) {
        if (i % 2 === 0) {
            finalResult = add(finalResult, i);
        } else {
            finalResult = subtract(finalResult, i);
        }
    }
    return finalResult;
}

// mutant imcd
function complexMathCalculation2(a, b, c, d) {
    const sum = 100;
    const difference = subtract(c, d);
    const product = multiply(sum, difference);
    const result = divide(product, 2);
    let finalResult = result;
    
    for (let i = 0; i < 5; i++) {
        if (i % 2 === 0) {
            finalResult = add(finalResult, i);
        } else {
            finalResult = subtract(finalResult, i);
        }
    }
    return finalResult;
}

// mutant irem
function complexMathCalculation3(a, b, c, d) {
    const sum = add(a, b);
    const difference = subtract(c, d);
    const product = multiply(sum, difference);
    const result = divide(product, 2);
    let finalResult = result;
    
    for (let i = 0; i < 5; i++) {
        if (i % 2 === 0) {
            finalResult = add(finalResult, i);
        } else {
            finalResult = subtract(finalResult, i);
        }
    }
    return -finalResult;
}

export { add, subtract, multiply, divide, complexMathCalculation, complexMathCalculation1, complexMathCalculation2, complexMathCalculation3 };

File: python/src/integrationSrc.py
def add(a, b):
    return a + b

def subtract(a, b):
    return a - b

def multiply(a, b):
    return a * b

def divide(a, b):
    if b == 0:
        raise ValueError("Division by zero is not allowed.")
    return a / b

def complex_math_calculation(a, b, c, d):
    sum_result = add(a, b)
    difference = subtract(c, d)
    product = multiply(sum_result, difference)
    result = divide(product, 2)
    final_result = result
    
    for i in range(5):
        if i % 2 == 0:
            final_result = add(final_result, i)
        else:
            final_result = subtract(final_result, i)
    
    return final_result

def complex_math_calculation1(d, b, c, a):
    sum_result = add(a, b)
    difference = subtract(c, d)
    product = multiply(sum_result, difference)
    result = divide(product, 2)
    final_result = result
    
    for i in range(5):
        if i % 2 == 0:
            final_result = add(final_result, i)
        else:
            final_result = subtract(final_result, i)
    
    return final_result

def complex_math_calculation2(a, b, c, d):
    sum_result = 100
    difference = subtract(c, d)
    product = multiply(sum_result, difference)
    result = divide(product, 2)
    final_result = result
    
    for i in range(5):
        if i % 2 == 0:
            final_result = add(final_result, i)
        else:
            final_result = subtract(final_result, i)
    
    return final_result

def complex_math_calculation3(a, b, c, d):
    sum_result = add(a, b)
    difference = subtract(c, d)
    product = multiply(sum_result, difference)
    result = divide(product, 2)
    final_result = result
    
    for i in range(5):
        if i % 2 == 0:
            final_result = add(final_result, i)
        else:
            final_result = subtract(final_result, i)
    
    return -final_result